app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/MultipleAccountsCheckController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers
+
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.config.AppConfig
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.connectors.EACDConnector
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.auth.AuthAction
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.errors.UnexpectedResponseFromEACD
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.views.html.UnderConstructionView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class MultipleAccountsCheckController @Inject()(
+  authAction: AuthAction,
+  mcc: MessagesControllerComponents,
+  eacdConnector: EACDConnector,
+  underConstructionView: UnderConstructionView
+)(implicit ec: ExecutionContext,
+  implicit val config: AppConfig)
+    extends FrontendController(mcc)
+      with I18nSupport {
+
+
+  def personalTaxEnrolmentCheck(redirectUrl: String): Action[AnyContent] = authAction.async {
```

Comment:
This function is going to be for all multipleAccounts so it should be named accordingly. i.e multipleAccountsCheck